osu.Game/Graphics/Backgrounds/BeatmapBackgroundWithStoryboard.cs

```diff
@@ -30,8 +38,33 @@ private void load()
             {
                 RelativeSizeAxes = Axes.Both,
                 Volume = { Value = 0 },
-                Child = new DrawableStoryboard(Beatmap.Storyboard) { Clock = new InterpolatingFramedClock(Beatmap.Track) }
+                Child = new DrawableStoryboard(Beatmap.Storyboard) { Clock = storyboardClock = new InterpolatingFramedClock(Beatmap.Track) }
             }, AddInternal);
         }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+            if (musicController != null)
+                musicController.TrackChanged += onTrackChanged;
+        }
+
+        private void onTrackChanged(WorkingBeatmap newBeatmap, TrackChangeDirection _)
+        {
+            if (newBeatmap != Beatmap)
+                return;
+
+            // `MusicController` will sometimes reload the track, even when the working beatmap technically hasn't changed.
+            // ensure that the storyboard's clock is always using the latest track instance.
+            storyboardClock.ChangeSource(newBeatmap.Track);
+            storyboardClock.ProcessFrame();
```

Comment:
What does the `ProcessFrame` do here?

Reply:
The clock is interpolating, so without an explicit `ProcessFrame()` call, the interpolation will get stuck at the time instant at which the clock was suspended before the source change. This manifests in the storyboard/video getting stuck at the instant it was suspended at and only resuming playback after the new beatmap track has caught up to that time instant.

I suppose that would require an inline comment as an explanation...?